File: app/src/main/java/me/zhanghai/android/files/provider/root/RootPosixFileAttributeView.java
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.root;

import androidx.annotation.NonNull;
import me.zhanghai.android.files.provider.common.PosixFileAttributeView;
import me.zhanghai.android.files.provider.common.PosixFileAttributes;
import me.zhanghai.android.files.provider.remote.RemoteInterfaceHolder;
import me.zhanghai.android.files.provider.remote.RemotePosixFileAttributeView;

public class RootPosixFileAttributeView<FA extends PosixFileAttributes>
        extends RemotePosixFileAttributeView<FA> {

    public RootPosixFileAttributeView(@NonNull PosixFileAttributeView attributeView) {
        super(new RemoteInterfaceHolder<>(() -> RootFileService.getInstance()
                .getRemotePosixFileAttributeViewInterface(attributeView)));
    }

    @Override
    public String name() {
        throw new AssertionError();
    }
}

File: app/src/main/java/me/zhanghai/android/files/provider/root/RootPosixFileAttributeView.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.root

import me.zhanghai.android.files.provider.common.PosixFileAttributeView
import me.zhanghai.android.files.provider.remote.RemoteInterface
import me.zhanghai.android.files.provider.remote.RemotePosixFileAttributeView

open class RootPosixFileAttributeView(
    attributeView: PosixFileAttributeView
) : RemotePosixFileAttributeView(
    RemoteInterface { RootFileService.getRemotePosixFileAttributeViewInterface(attributeView) }
) {
    override fun name(): String {
        throw AssertionError()
    }
}
